File: src/sdk/v2/python/examples/bridge_operations.py
import asyncio
from vernachain_sdk import VernachainClient
from vernachain_sdk.models import BridgeTransferRequest

async def main():
    async with VernachainClient(
        node_url="http://localhost:8545",
        api_key="your-api-key"
    ) as client:
        try:
            # Create a bridge transfer to Ethereum
            transfer_request = BridgeTransferRequest(
                target_chain="ethereum",
                amount=5.0,
                recipient="0xETH_ADDRESS...",
                gas_limit=100000
            )
            
            # Initiate the bridge transfer
            transfer = await client.bridge_transfer(transfer_request)
            print(f"Bridge transfer initiated: {transfer.transfer_id}")
            print(f"From: {transfer.sender} (Vernachain)")
            print(f"To: {transfer.recipient} (Ethereum)")
            print(f"Amount: {transfer.amount}")
            print(f"Status: {transfer.status}")
            
            # Monitor the transfer status
            while transfer.status not in ["completed", "failed"]:
                await asyncio.sleep(10)  # Check every 10 seconds
                
                # In a real application, you would want to:
                # 1. Get the transfer status from both chains
                # 2. Verify the proof of transfer
                # 3. Handle any required confirmations
                print("\nChecking transfer status...")
                print(f"Current status: {transfer.status}")
                
                if transfer.completed_at:
                    print(f"Completed at: {transfer.completed_at}")
                    if transfer.proof:
                        print("Transfer verified with proof")
                        print(f"Transaction hash on target chain: {transfer.proof.get('target_tx_hash')}")
                
            print("\nFinal transfer status:")
            print(f"Status: {transfer.status}")
            if transfer.status == "completed":
                print("Transfer successfully completed!")
                print(f"Time taken: {transfer.completed_at - transfer.initiated_at}")
            else:
                print("Transfer failed. Please check the error details.")

        except Exception as e:
            print(f"Error: {e}")

if __name__ == "__main__":
    asyncio.run(main()) 

File: src/sdk/v2/php/examples/bridge-operations.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

use Vernachain\SDK\VernachainClient;

try {
    $client = new VernachainClient([
        'node_url' => 'http://localhost:8545',
        'api_key' => 'your-api-key'
    ]);

    // Lock tokens in source chain
    $lockRequest = [
        'token_address' => '0x1234...', // ERC20 token address
        'amount' => '1000000000000000000', // 1 token with 18 decimals
        'target_chain' => 'ethereum',
        'recipient' => '0x5678...', // recipient address on target chain
        'shard_id' => 0,
        'gas_price' => 1.5,
        'gas_limit' => 100000
    ];

    $lockResult = $client->lockTokens($lockRequest);
    echo "Tokens locked. Transaction hash: {$lockResult['transaction_hash']}\n";
    echo "Bridge transfer ID: {$lockResult['bridge_transfer_id']}\n";

    // Monitor bridge transfer status
    $bridgeStatus = $client->getBridgeTransferStatus($lockResult['bridge_transfer_id']);
    while (!in_array($bridgeStatus['status'], ['completed', 'failed'])) {
        echo "Bridge transfer status: {$bridgeStatus['status']}\n";
        echo "Confirmations: {$bridgeStatus['confirmations']}/{$bridgeStatus['required_confirmations']}\n";
        
        sleep(10); // Wait 10 seconds
        $bridgeStatus = $client->getBridgeTransferStatus($lockResult['bridge_transfer_id']);
    }

    echo "Final bridge transfer status: {$bridgeStatus['status']}\n";
    if ($bridgeStatus['status'] === 'completed') {
        echo "Target chain transaction hash: {$bridgeStatus['target_transaction_hash']}\n";
    }

    // Example of unlocking tokens (receiving from another chain)
    $pendingTransfers = $client->getPendingBridgeTransfers('ethereum');
    foreach ($pendingTransfers as $transfer) {
        if ($transfer['status'] === 'pending_unlock') {
            $unlockResult = $client->unlockTokens([
                'bridge_transfer_id' => $transfer['bridge_transfer_id'],
                'proof' => $transfer['proof'],
                'shard_id' => 0,
                'gas_price' => 1.5,
                'gas_limit' => 150000
            ]);
            echo "Tokens unlocked. Transaction hash: {$unlockResult['transaction_hash']}\n";
        }
    }

} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
} 